nupic/embodied/utils/garage_utils.py

```diff
@@ -36,194 +36,90 @@
-          [b_21, b_22, b_23, ... b_2n],
-          ...
-          [b_m1, b_m2, b_m3, ... b_mn] ]
-        rewards:
-        [ [r_11, r_12, r_13, ... r_1n],
-          [r_21, r_22, r_23, ... r_2n],
-          ...
-          [r_m1, r_m2, r_m3, ... r_mn] ]
-
-    Args:
-        discount (float): RL discount factor (i.e. gamma).
-        gae_lambda (float): Lambda, as used for Generalized Advantage
-            Estimation (GAE).
-        max_episode_length (int): Maximum length of a single episode.
-        baselines (torch.Tensor): A 2D vector of value function estimates with
-            shape (N, T), where N is the batch dimension (number of episodes)
-            and T is the maximum episode length experienced by the agent. If an
-            episode terminates in fewer than T time steps, the remaining
-            elements in that episode should be set to 0.
-        rewards (torch.Tensor): A 2D vector of per-step rewards with shape
-            (N, T), where N is the batch dimension (number of episodes) and T
-            is the maximum episode length experienced by the agent. If an
-            episode terminates in fewer than T time steps, the remaining
-            elements in that episode should be set to 0.
-
-    Returns:
-        torch.Tensor: A 2D vector of calculated advantage values with shape
-            (N, T), where N is the batch dimension (number of episodes) and T
-            is the maximum episode length experienced by the agent. If an
-            episode terminates in fewer than T time steps, the remaining values
-            in that episode should be set to 0.
-
-    """
-    adv_filter = torch.full((1, 1, 1, max_episode_length - 1),
-                            discount * gae_lambda,
-                            dtype=torch.float, device=global_device())
-    adv_filter = torch.cumprod(F.pad(adv_filter, (1, 0), value=1), dim=-1)
-
-    deltas = (rewards + discount * F.pad(baselines, (0, 1))[:, 1:] - baselines)
-    deltas = F.pad(deltas,
-                   (0, max_episode_length - 1)).unsqueeze(0).unsqueeze(0)
-
-    advantages = F.conv2d(deltas, adv_filter, stride=1).reshape(rewards.shape)
-    return advantages
+from nupic.embodied.multitask.dendrites import AbsoluteMaxGatingDendriticLayer
 
 
 def create_policy_net(env_spec, net_params):
-    net_type = net_params.net_type
-    assert net_type in {"MLP", "Dendrite_MLP"}
-    if net_type == "MLP":
+    if net_params.net_type == "MLP":
         net = GaussianMLPPolicy(
             env_spec=env_spec,
-            hidden_sizes=net_params.policy_hidden_sizes,
+            hidden_sizes=net_params.hidden_sizes,
             hidden_nonlinearity=create_nonlinearity(net_params.policy_hidden_nonlinearity),
-            output_nonlinearity=create_nonlinearity(net_params.policy_output_nonlinearity),
-            min_std=net_params.policy_min_std,
-            max_std=net_params.policy_max_std,
+            output_nonlinearity=create_nonlinearity(net_params.output_nonlinearity),
```

Comment:
Important note: in general, for all projects, whenever we change in the API all experiment files should be adjusted accordingly to maintain compatibility.

Reply:
Yep, already did that 🙂